vector-im/element-android
# FTUE - Sign up analytics

## Type of change


## Content

Fixes #5285 

Adds `SignUp` event tracking after the user creates an account **AND** consents to tracking. 

- Creates an `AuthenticationDescription` abstraction  and also sets the ground work for tracking sign in in the future.
- Chains the `AuthenticationDescription` through to the `HomeActivity` in order to be tracked

## Motivation and context

To enable the collection of sign up metrics (and sign up type)

## Screenshots / GIFs

No UI changes

```
D/Analytics: capture(Signup(authenticationType=Password))
```

![2022-05-13T09:09:08,565852105+01:00](https://user-images.githubusercontent.com/1848238/168239970-8cb7f232-b196-4787-a49a-5f817cd61fca.png)

## Tests

- With the combined login feature flag enabled
- Create an account
- Notice the SignUp tracking event in posthog

## Tested devices

- OS version(s): 28
